codegen/config.go

```diff
@@ -191,6 +191,47 @@ func (cfg *Config) Check() error {
 	if err := cfg.Resolver.Check(); err != nil {
 		return errors.Wrap(err, "config.resolver")
 	}
+
+	// check packages names against conflict, if present in the same dir
+	// and check filenames for uniqueness
+	packageConfigList := []PackageConfig{
+		cfg.Model,
+		cfg.Exec,
+		cfg.Resolver,
+	}
+	filesMap := make(map[string]bool)
+	pkgConfigsByDir := make(map[string][]PackageConfig)
+	for i, current := range packageConfigList {
+		if i == 0 {
+			filesMap[current.Filename] = true
+			pkgConfigsByDir[current.Dir()] = []PackageConfig{current}
+			continue
+		}
+		_, fileFound := filesMap[current.Filename]
+		if fileFound {
+			return fmt.Errorf("filename %s defined more than once", current.Filename)
+		}
+		filesMap[current.Filename] = true
+		prevPkgList, inSameDir := pkgConfigsByDir[current.Dir()]
+		if inSameDir {
+			for _, previous := range prevPkgList {
+				if current.Package != previous.Package {
+					eitherPackageEmpty := previous.Package != "" || current.Package != ""
+					if eitherPackageEmpty {
+						if current.Package == filepath.Base(current.Dir()) && previous.Package == "" {
```

Comment:
This is reproducing logic from https://github.com/99designs/gqlgen/blob/next/codegen/config/config.go#L138
which is called by c.Exec.Check() earlier in this function

Maybe its best to let normalise run first?

Reply:
Exactly. I think config.Check() should be called after it's been normalized.
I'll see how that will work.